test/extended/util/ldap.go

```diff
@@ -0,0 +1,200 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"time"
+
+	ldap "gopkg.in/ldap.v2"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	appsv1 "github.com/openshift/api/apps/v1"
+	"github.com/openshift/library-go/pkg/config/helpers"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/origin/test/extended/testdata"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/wait"
+)
+
+// CreateLDAPTestServer deploys an LDAP server, returning the server address or err.
+func CreateLDAPTestServer(oc *CLI) (string, error) {
+	deploy, service, ldif, scripts, err := readLDAPServerTestData()
+	if err != nil {
+		return "", err
+	}
+
+	certDir, err := ioutil.TempDir("", "testca")
+	if err != nil {
+		return "", err
+	}
+
+	ca, err := crypto.MakeSelfSignedCA(path.Join(certDir, "ca.crt"), path.Join(certDir, "ca.key"), path.Join(certDir, "serial"), "ldap CA", 100)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = oc.AdminKubeClient().CoreV1().ConfigMaps(oc.Namespace()).Create(ldif)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = oc.AdminKubeClient().CoreV1().ConfigMaps(oc.Namespace()).Create(scripts)
+	if err != nil {
+		return "", err
+	}
+
+	createdService, err := oc.AdminKubeClient().CoreV1().Services(oc.Namespace()).Create(service)
+	if err != nil {
+		return "", err
+	}
+
+	err = oc.AsAdmin().Run("expose").Args("service", service.Name).Execute()
+	if err != nil {
+		return "", err
+	}
+
+	route, err := oc.AdminRouteClient().Route().Routes(oc.Namespace()).Get(service.Name, v1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	serverCertConfig, err := ca.MakeServerCert(sets.NewString("localhost", route.Spec.Host, createdService.Spec.ClusterIP), 100)
+	if err != nil {
+		return "", err
+	}
+
+	serverCertPEM, serverCertKeyPEM, err := serverCertConfig.GetPEMBytes()
+	if err != nil {
+		return "", err
+	}
+
+	_, err = oc.AdminKubeClient().CoreV1().Secrets(oc.Namespace()).Create(&corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ldap-cert",
+		},
+		Data: map[string][]byte{
+			corev1.TLSCertKey:       serverCertPEM,
+			corev1.TLSPrivateKeyKey: serverCertKeyPEM,
+		},
+		Type: corev1.SecretTypeTLS,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Create SA
+	_, err = oc.AdminKubeClient().CoreV1().ServiceAccounts(oc.Namespace()).Create(&corev1.ServiceAccount{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ldap",
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Allow the openldap container to run as root. This lets us use the existing openldap server container startup scripts to poplate its database using ldapi:///.
+	err = oc.AsAdmin().Run("create").Args("role", "scc-anyuid", "--verb=use", "--resource=scc", "--resource-name=anyuid").Execute()ru
```

Comment:
followup: I prefer creating real objects and using real clients, so creating the rolebiding you want.